### PR TITLE
Add IAS targetting optimisations when defining ad slots.

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
IAS periodically scan our inventory and build models that describe the average viewability, brand safety and attention that various slots on various sections receive.

We can query their optimisation tool to request this information prior to requesting an ad from DFP. We can then use this information to add targeting to the ad slot beforehand.

Whilst IAS supports the batch requesting of ad slot information, we send a single request to IAS for _each_ slot that is to be defined. Because we lazy load and define slots all over the place at different times, there is no sensible bottleneck in which to insert this behaviour and thus we must do it on each creation of an `Advert` in `define-slot`.

**Note: this currently doesn't work. The timeout is always reached, however for IAS to debug this it needs to be present on a test page i.e. CODE. Instead of commandeering code for an interminable length of time, I plan to merge this and only enable the switch on CODE.**

## What is the value of this and can you measure success?
We can better target campaigns in DFP with richer information about the resultant, displayed ad.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/define-slot.js [hunk 1]
```diff
@@ -96,6 +96,56 @@ const defineSlot = (adSlotNode: Element, sizes: Object): Object => {
         slotReady = setHighMerchSlotTargeting(slot, slotTarget);
     }
 
+    if (config.switches.iasOptimisation) {
+        console.log(`Defining ${id}`);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: remove console.log

File: static/src/javascripts/projects/commercial/modules/dfp/define-slot.js [hunk 2]
```diff
@@ -96,6 +96,56 @@ const defineSlot = (adSlotNode: Element, sizes: Object): Object => {
...
+        iasPET.pubId = '10249';
+
+        let loadedResolve;
+
+        const iasDataPromise = new Promise(resolve => {
+            loadedResolve = resolve;
+        });
+
+        const iasDataCallback = () => {
+            console.log(`Applying IAS targeting: ${id}`);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: another console log

**Comment by rich-nguyen** [comment]
As discussed, an explicit check for the CODE stage would be safer here. I'd also add comments to the console logs to explain they are for CODE only.

**Review by rich-nguyen — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/define-slot.js (hunk 1 above)
Comment (JonNorman) [reply static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: As discussed, these are being left in to help better troubleshoot this problem on CODE.

File: static/src/javascripts/projects/commercial/modules/dfp/define-slot.js (hunk 2 above)
Comment (JonNorman) [reply static/src/javascripts/projects/commercial/modules/dfp/define-slot.js]: As discussed in person, these are being left in to help better troubleshoot this problem on CODE.

**Pull request merged**